packages/esm-form-entry-app/src/app/openmrs-api/obs-resource.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpParams } from '@angular/common/http';
+
+import { map } from 'rxjs/operators';
+import { WindowRef } from '../window-ref';
+import { fhirBaseUrl } from '@openmrs/esm-framework';
+
+@Injectable()
+export class ObservationResourceService {
+  constructor(protected http: HttpClient, protected windoRef: WindowRef) {}
+
+  public getMostRecentObsValues(date, conceptUuids, patientUuid) {
+    const url = `/openmrs${fhirBaseUrl}/Observation/$lastn`;
```

Comment:
Don't hard-code `/openmrs` We actually support this being changed. Just use the `makeUrl` function exported from esm-framework, i.e., 

```
makeUrl(`${fhirBaseUrl}/Observation/$lastn`);
```

Reply:
Actually, though, is there a reason we can't use `openmrsFetch` here? I'd always prefer using that unless it's impossible to do so... (`openmrsFetch` handles a number of concerns...).

Reply:
I am not sure on how to use `openmrsFetch` on the angular side, from what I have seen in the existing services, they mostly use the approach I used, but I would also prefer using openmrsFetch, I will try it and see.

Reply:
You should just be able to import it from `@openmrs/esm-framework`... I guess there's a conundrum about how the form will access the results... `openmrsFetch` is Promise-based; Angular's HTTP framework is built on rxjs... Neither of those return straight-forward values.

Reply:
We might have to use XMLHttpRequest for this...

Reply:
If all that we need is an observable we can just use rxjs for this:

```js
import { from } from 'rxjs';
const observable = from(promise);
```

but I don't know why the given approach was selected in favor of `openmrsFetch`.